libs/sdk-core/src/greenlight/node_api.rs

```diff
@@ -1435,6 +1435,67 @@ impl NodeAPI for Greenlight {
         debug!("send_custom_message returned status {:?}", resp.status);
         Ok(())
     }
+
+    // Gets the routing hints related to all private channels that the node has
+    async fn get_routing_hints(&self) -> NodeResult<Vec<RouteHint>> {
```

Comment:
This can potentially return many routing hints. I think we'd want to select a maximum of 3 hints. Probably ordered by the liquidity available (if possible). 

Note that a route hint is (usually) not useful if the peer only has private channels as well. So you might want to select nodes with active public channels in the lightning graph.

Reply:
Do you see any real issue by leaving the logic trivial?  I don't want to complicate it if not really necessary.

Reply:
Not necessarily. Chances are high greenlight will already have hop hints in the invoice. Just a limit of 2 or 3 hints would be good.

Reply:
I think the issue is GL is not reliably adding the hop hints, especially when creating an invoice very soon after connecting

Reply:
Added limit